tardis/montecarlo/src/test_cmontecarlo.c

```diff
@@ -0,0 +1,239 @@
+double test_compute_distance2line(void);
+double test_compute_distance2electron(void);
+double test_rpacket_doppler_factor(void);
+double test_move_packet(void);
+int64_t test_montecarlo_one_packet(void);
+int64_t test_montecarlo_one_packet_loop(void);
+bool test_montecarlo_line_scatter(void);
+double test_increment_j_blue_estimator(void);
+bool test_montecarlo_thomson_scatter(void);
+
+/* initialise RPacket */
+void
+init_rpacket(void){
+	rp = (rpacket_t *) malloc(sizeof(rpacket_t));
+	
+	double MU = 0.3;
+	double R = 7.5e14;
+	double ENERGY = 0.9;
+	int NEXT_LINE_ID = 1;
+	double NU = 0.4;
+	double NU_LINE = 0.2;
+	int CURRENT_SHELL_ID = 0;
+
+	double TAU_EVENT = 2.9e13;
+
+	rpacket_set_current_shell_id(rp, CURRENT_SHELL_ID);
+	rpacket_set_next_shell_id(rp, CURRENT_SHELL_ID+1);
+	rpacket_set_mu(rp, MU);
+	rpacket_set_nu(rp, NU);
+	rpacket_set_r(rp, R);
+	rpacket_set_last_line(rp, false);
+	rpacket_set_recently_crossed_boundary(rp, 1);
+
+	rpacket_set_close_line(rp, false);
+	rpacket_set_nu_line(rp, NU_LINE);
+
+	rpacket_set_next_line_id(rp, NEXT_LINE_ID);
+
+	rpacket_set_tau_event(rp, TAU_EVENT);
+	rpacket_set_virtual_packet(rp, 0);
+	rpacket_set_energy(rp, ENERGY);
+	rpacket_set_virtual_packet_flag(rp, true);
+	rpacket_set_status(rp, TARDIS_PACKET_STATUS_IN_PROCESS);
+
+	/*
+	rpacket_set_virtual_packet(rp, 1);
+	rpacket_set_current_shell_id(rp, 1);
+	rpacket_set_tau_event(rp, TAU_EVENT);
+	rpacket_set_d_electron(rp, D_ELECTRON);
+	*/
+
+}
+
+/* initialise storage model */
+void
+init_storage_model(void){
+	sm = (storage_model_t *) malloc(sizeof(storage_model_t));
+	int NUMBER_OF_SHELLS = 2;
+
+	sm->time_explosion = TIME_EXPLOSION;
+	sm->inverse_time_explosion = 1.0/TIME_EXPLOSION;
+	sm->inverse_sigma_thomson = 1.0/SIGMA_THOMSON;
+
+	/* R_OUTER = {8.64e14, 1.0368e15} */
+	sm->r_outer = (double *) malloc(sizeof(double)*NUMBER_OF_SHELLS);
+	sm->r_outer[0] = 8.64e14;
+	sm->r_outer[1] = 1.0368e15;
+
+	/* R_INNER = {6.912e14, 8.64e14} */
+	sm->r_inner = (double *) malloc(sizeof(double)*NUMBER_OF_SHELLS);
+	sm->r_inner[0] = 6.912e14;
+	sm->r_inner[1] = 8.64e14;
+	sm->no_of_lines = 2;
+
+	/* LINE_LIST_NU = { 1.26318289e+16,  1.26318289e+16,
+						1.23357675e+16,  1.23357675e+16, 
+						1.16961598e+16 };
+	*/
+	sm->line_list_nu = (double *) malloc(sizeof(double)*5);
+	sm->line_list_nu[0] = 1.26318289e+16;
+	sm->line_list_nu[1] = 1.26318289e+16;
+	sm->line_list_nu[2] = 1.23357675e+16;
+	sm->line_list_nu[3] = 1.23357675e+16;
+	sm->line_list_nu[4] = 1.16961598e+16;
+
+	/* INVERSE_ELECTRON_DENSITIES = {} */
+	sm->inverse_electron_densities = (double *) malloc(sizeof(double)*NUMBER_OF_SHELLS);
+	sm->inverse_electron_densities[0] = 1e-9;
+	sm->inverse_electron_densities[1] = 1e-9;
+	sm->electron_densities = (double *) malloc(sizeof(double )*2);
+	sm->electron_densities[0] = 1e9;
+	sm->electron_densities[1] = 1e9;
+
+
+	/* The following values need to be 
+	*  reassigned properly. By default
+	*  I assigned all 0.
+	*/
```

Comment:
@kaushik94 They're in the doc now. Most of it wasn't in the atomic data, but I've given you reasonable values (I think). Let me know if you have any more problems.